akka-projection-kafka/src/main/scala/akka/projection/kafka/internal/KafkaSourceProviderImpl.scala

```diff
@@ -104,6 +107,13 @@ import org.apache.kafka.common.TopicPartition
         "The offset contains Kafka topic partitions that were revoked or lost in a previous rebalance")
   }
 
+  override def extractCreationTime(record: ConsumerRecord[K, V]): Long = {
+    if (record.timestampType() == TimestampType.CREATE_TIME)
+      record.timestamp()
+    else
+      0L
```

Comment:
0L is a valid timestamp as a well as any Long value. This is give you the impression that we have a huge lag.

I lobby for removing the `Option` myself, but didn't think about the impact of using 0L (see https://github.com/akka/akka-projection/pull/265#discussion_r444450841).

Reply:
I think telemetry could treat `0L` as a special case, since it's a near impossibility to be a valid timestamp these days, but I think a `0L` might aid in troubleshooting something else, like a custom overridden timestamp (that's not implemented properly).  Should we use `-1L` instead?

Reply:
I agree that `0L` although possible it's not realistic. 

`-1L` is indeed safer, but we must stay with `0L`. The reason being that then the timestamp is not available in the akka-persistence journal, we use `0L`. We need to stay with it.

Reply:
As said, we already use 0L as the magic for Persistence plugins that don't implement the timestamp.
Option gives unnecessary overhead with extra allocation for each envelope.